internal/controllers/machine/machine_controller.go

```diff
@@ -488,6 +499,74 @@ func (r *Reconciler) isDeleteNodeAllowed(ctx context.Context, cluster *clusterv1
 	return nil
 }
 
+// isVersionAllowed returns true if the machine version conforms to the Kubernetes version skew policy.
+// If the Machine is part of the control plane, then its major.minor version must be greater than or equal to the
+// control plane actual major.minor version, and less than or equal to the control plane desired major.minor version.
+// If the Machine is not part of the control plane, then its major.minor version must be less than or equal to the control
+// plane actual major.minor version.
+func (r *Reconciler) isVersionAllowed(ctx context.Context, machine *clusterv1.Machine, cluster *clusterv1.Cluster) (bool, error) {
```

Comment:
I wonder if it would make sense to align some of the logic with what we are doing in the topology reconciler when triggering MD rollouts (computeMachineDeploymentVersion).

E.g. when the control plane is currently upgrading (ControlPlaneContract.IsUpgrading) we are not triggering new MD upgrades.

Most of the logic does not apply, but I wonder if there are some common parts.